### PR TITLE
[Crafting] Cleanup GP Vendors (Union Representatives)

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Cleans union representative NPCs via the following changes:
- Renames their functions
- Moves their inventories to crafting.lua
- Renames the variables used in their functions.

Additionally, this also has the following changes in the fishing guild:
- Minor correction in the required rank for an item.
- Fixes items selected not matching items given.

A note on Guild IDs: We use 2 different guild id systems. This NPCs use the correct order used by retail. However, the order we have defined doesn't match it and I don't see why this order was chosen, since it also doesn't follow the skill ID order.

Once I get to cleaning guild masters, I will clean those up, but, for the time being, that's out of the scope of this PR.
BUT it WILL get done.

## Steps to test these changes

Review by commit for clarity (maybe)
It should be functionally the same. Except for the fishing guild, which should work properly now.

### CODE REVIEW

**Comment by Xaver-DaRed** [comment]
Tested it again and screenshoted the result.
imagen

File: scripts/globals/crafting.lua [hunk 1]
```diff
@@ -119,7 +321,7 @@ end
 -- tradeTestItem Action
 -----------------------------------
 xi.crafting.tradeTestItem = function(player, npc, trade, craftID)
-    local guildID    = craftID - 48
+    local guildId    = craftID - 48
```
Comment (TeoTwawki) [inline scripts/globals/crafting.lua:322]: I know we normally fullLowerCamel, but I think an exception can be made for `ID` given its already short for "**Id**entification"

I really hate seeing `Id` anywhere. Just triggers the heck outa my OCD.

Comment (Xaver-DaRed) [reply scripts/globals/crafting.lua:322]: We were using both `ID` and `Id` in this file. I have no real preference.

Comment (Xaver-DaRed) [reply scripts/globals/crafting.lua:322]: I have a new, separate PR half ready touching this file (Guild Masters in this case)

If you guys want `ID` I promise to change it in that one.

**Comment by Xaver-DaRed** [comment]
Rebased to resolve conflicts

File: scripts/globals/crafting.lua [hunk 2]
```diff
@@ -244,74 +446,79 @@ xi.crafting.unionRepresentativeTrade = function(player, npc, trade, csid, guildI
     end
 end
 
-xi.crafting.unionRepresentativeTrigger = function(player, guildId, csid, currency, keyitems)
+xi.crafting.guildPointNPConTrigger = function(player, csid, guildId)
```
Comment (claywar) [inline scripts/globals/crafting.lua]: I'd prefer if we're calling this onTrigger to pass the two expected parameters (player, npc) to the function, and doing a lookup from a table in this script to determine what the guildId and other values are.

Comment (claywar) [reply scripts/globals/crafting.lua]: (Also, I can't help but read this as NP-Con)

**Review by claywar — APPROVED** [review]
Definitely an improvement overall, but I'd like us to move the passed values from the NPC scripts to a lookup in the global eventually.

File: scripts/globals/crafting.lua (hunk 2 above)
Comment (Xaver-DaRed) [reply scripts/globals/crafting.lua]: I can change the parameters in the next PR, once I actually know all guild-related stuff that can be tabled.
The consensus back then was that "Union Representative" was not really known/clear.
Would you preffer `guildPointOnTrigger`?

Comment (claywar) [reply scripts/globals/crafting.lua]: I can get on board with this

**Pull request merged**